keras/preprocessing/image.py

```diff
@@ -142,18 +142,21 @@ def __init__(self,
     def flow(self, X, y, batch_size=32, shuffle=False, seed=None,
              save_to_dir=None, save_prefix="", save_format="jpeg"):
         assert len(X) == len(y)
-        if seed:
-            random.seed(seed)
-
-        if shuffle:
-            seed = random.randint(1, 10e6)
-            np.random.seed(seed)
-            np.random.shuffle(X)
-            np.random.seed(seed)
-            np.random.shuffle(y)
 
         b = 0
+        total_b = 0
         while 1:
+            if b == 0:
+                if seed is not None:
+                    random.seed(seed + total_b)
+
+                if shuffle:
+                    seed = random.randint(1, 10e6)
+                    np.random.seed(seed)
+                    np.random.shuffle(X)
+                    np.random.seed(seed)
+                    np.random.shuffle(y)
```

Comment:
More elegant would be to shuffle an array of indices once, then use the array to index X and y.